markdown/ch8.markdown

```diff
@@ -30,16 +30,12 @@ print(sum_of_means)
 This strategy works really well for cases where your problem can be neatly solved with by-row processing.
 But things can quickly get out of hand when your code becomes more sophisticated.
 
-In the most commonly used programming models, you pass your data to a function,
-the function processes the data, and then returns the result. Done. (This
-includes most of this book!)
-
-But in streaming programs, a function processes *some* of the data, returns the
+In streaming programs, a function processes *some* of the input data, returns the
 processed chunk, then, while downstream functions are dealing with that chunk,
 the function receives a bit more, and so on...  All these things are going on
 at the same time!  How can one keep them straight?
 
-We too found this difficult to do, until we found the `toolz` library.
+We too found this difficult, until we found the `toolz` library.
```

Comment:
found...found.  How about "We too found this difficult, until discovering the `tools` library."